pkg/databind/internal/secrets/cyberarkcli.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package secrets
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/databind/pkg/data"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+	"os/exec"
+	"strings"
+)
+
+// Make mocking simpler
+var cyberArkExecCommand = exec.Command
+
+type CyberArkCLI struct {
+	CLI    string `yaml:"cli"`
+	AppID  string `yaml:"app-id"`
+	Safe   string
+	Folder string
+	Object string
+}
+
+type cyberArkCLIGatherer struct {
+	cfg *CyberArkCLI
+}
+
+// CyberArkCLIGatherer instantiates a CyberArkCLI variable gatherer from the given configuration.
+// The result is a map with a single "password" key value pair
+func CyberArkCLIGatherer(cyberArkCLI *CyberArkCLI) func() (interface{}, error) {
+	g := cyberArkCLIGatherer{cfg: cyberArkCLI}
+	return func() (interface{}, error) {
+		dt, err := g.get()
+		if err != nil {
+			return "", err
+		}
+		return dt, err
+	}
+}
+
+func (g *cyberArkCLIGatherer) get() (data.InterfaceMap, error) {
+	cmd := cyberArkExecCommand(g.cfg.CLI, "GetPassword", "-p", "AppDescs.AppID="+g.cfg.AppID, "-p", "Query=Safe="+g.cfg.Safe+";Folder="+g.cfg.Folder+";Object="+g.cfg.Object, "-o", "Password")
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve cyberArkCLI secret from cli. err: %s err msg: %s", err, stderr.String())
+	}
+	password := strings.TrimSuffix(out.String(), "\n")
+	password = strings.TrimSuffix(out.String(), "\r")
```

Comment:
Should it be?
```
	password := strings.TrimSuffix(out.String(), "\n")
	password = strings.TrimSuffix(password, "\r")
```

Reply:
I wouldn't think so. Linux will add \n, Windows \r it will only ever be one or the other.

Reply:
But you will overwrite with the latest one. As I know if you want to sanitize both symbols, you have to apply both rules, even if one of them will be redundant on each system. 

You can see example here: https://golang.org/pkg/strings/#TrimSuffix

Reply:
I changed it to your way as it makes no difference.